### DISCUSSION TITLE
Heading tags in transclusions

### DISCUSSION THREAD

**Discussion by DeRuiterIndustrieleAutomatisering** [TiddlyWiki/TiddlyWiki5#6683] [Q&A]
When I define a structure like:
! Level 1
!! Level 2a
!! Level 2b

and a transclusion at level 2a with the content
! Trans Level 1
text text

It transcludes as 
! Level 1
!! Level 2a
**! Trans Level 1**
!! Level 2b

But is it possible to transclude this and in the same time change the heading level (depending on the position where the transclusion is placed)?

! Level 1
!! Level 2a
**!!!** Trans Level 1
!! Level 2b

**Comment by Jermolene** [comment]
Hi @DeRuiterIndustrieleAutomatisering, and welcome to the project.

You can use the `tv-adjust-heading-level` variable to set a base heading level which is added to that of the headings that are encountered.

You could use it like this:

```
! Level 1
!! Level 2a

<$let tv-adjust-heading-level="2"><$transclude tiddler="other-tiddler-title"/></$let>

!! Level 2b
```

See the [docs](https://tiddlywiki.com/#tv-adjust-heading-level%20Variable%20(Examples):%5B%5Btv-adjust-heading-level%20Variable%5D%5D%20%5B%5Btv-adjust-heading-level%20Variable%20(Examples)%5D%5D)

**Reply by DeRuiterIndustrieleAutomatisering** [reply]
hi Jeremy,
I'm a long time TW user and once a while I start searching for a replacement. But so far always come back to Tiddlywiki

I was wondering if the header level downgrading can be done automatically? Most of the times I have a longer story that I start breaking up in smaller parts by using ^-E (excise). After this the levels don't match in the new tiddler that I want to use for printing

**Reply by Jermolene** [reply]
> I'm a long time TW user and once a while I start searching for a replacement. But so far always come back to Tiddlywiki

Good to hear, thank you!

> I was wondering if the header level downgrading can be done automatically? Most of the times I have a longer story that I start breaking up in smaller parts by using ^-E (excise). After this the levels don't match in the new tiddler that I want to use for printing

Interesting idea. The idea would be to automatically set `tv-adjust-heading-level` to the current heading level before transcluding another tiddler.

The challenge is somewhat arcane: TiddlyWiki renders content as a tree with the constraint that information can only be inherited from parents to children, and not between siblings. That's also why we don't have any automatic numbering features.

Something that might work for you now would be to use a custom macro to do the transclusion, and give it an additional parameter for setting `tv-adjust-heading-level`. For example:

```
\define heading-adjusted-transclude(level,tiddler)
<$let tv-adjust-heading-level=<<__level__>>><$transclude tiddler=<<__tiddler__>> mode="block"/></$let>
\end

...
!! A heading

<<heading-adjusted-transclude 2 "My tiddler">>
```

You'd want to rename the macro to something supershort like `_t` for ease of typing.
